### PR TITLE
trying new ConanFile import location from conan import ConanFile

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Enable ``from conan import ConanFile`` to prepare for future namespace.
Docs: https://github.com/conan-io/docs/pull/2318

Just to discuss:
- New import?
- New ``class Recipe:`` convention?

### CODE REVIEW

**Comment by lasote** [comment]
I would say the new import has to exist, nothing should be left in "conans" BUT as the "opt-in" for possible breaking changes I like the `class Recipe` , it looks more friendly and explicit.

**Comment by memsharded** [comment]
> I would say the new import has to exist, nothing should be left in "conans" BUT as the "opt-in" for possible breaking changes I like the class Recipe , it looks more friendly and explicit.

Then, should we still make the ``from conan import ConanFile`` available? Maybe not, if the naming convention is good enough, I would go for it, as the way forward in 2.0, and avoid 2 different ways of doing the same. My only (minor) concern is a good naming and if that name could have potential conflicts (users having ``Recipe`` generic classes in their python_requires, for example)

Possible names:
- ``class Recipe:``
- ``class ConanRecipe:``
- ``class Conanfile``: # we can play here with the casing, ConanFile vs Conanfile, could work
- Any other idea?

**Comment by lasote** [comment]
As we talked, we could introduce the "class Recipe" as the new way to declare the recipe (without imports) but to keep compatibility, check first that there is no other conanfile subclass just in case we break some user naming the classes as "Recipe"

File: conans/client/loader.py [hunk 1]
```diff
@@ -373,6 +373,8 @@ class defining the Recipe, but also process possible existing generators
                 raise ConanException("More than 1 conanfile in the file")
         elif issubclass(attr, Generator) and attr != Generator:
             generator_manager.add(attr.__name__, attr, custom=True)
+        elif attr.__name__ == "Recipe":
+            result = type('Recipe', (attr, ConanFile, object), {})
```
Comment (memsharded) [inline conans/client/loader.py]: This seems to work, I have run some tests forcing ``GenConanfile`` to always use this way, and the only tests the break seems the ones that change the recipe revision (obviously, conanfile.py is different).

However, this feels a bit like too magic. I am creating a completely new type, ``Recipe`` that derives from the user ``Recipe`` class written in the ``conanfile.py`` (but still a different entity, I am not just injecting the ``ConanFile`` base class).

So not sure what would be the best approach:
- To use explicit ``from conan import ConanFile  & class Pkg(ConanFile)``
- To use magical ``class Recipe:``

**Pull request merged**